embedded-hal-bus/src/spi/mutex.rs

```diff
@@ -12,30 +13,48 @@ use super::DeviceError;
 /// Sharing is implemented with a `std` [`Mutex`](std::sync::Mutex). It allows a single bus across multiple threads,
 /// with finer-grained locking than [`CriticalSectionDevice`](super::CriticalSectionDevice). The downside is
 /// it is only available in `std` targets.
-pub struct MutexDevice<'a, BUS, CS> {
+pub struct MutexDevice<'a, BUS, CS, D> {
     bus: &'a Mutex<BUS>,
     cs: CS,
+    delay: D,
 }
 
-impl<'a, BUS, CS> MutexDevice<'a, BUS, CS> {
+impl<'a, BUS, CS, D> MutexDevice<'a, BUS, CS, D> {
     /// Create a new ExclusiveDevice
-    pub fn new(bus: &'a Mutex<BUS>, cs: CS) -> Self {
-        Self { bus, cs }
+    pub fn new(bus: &'a Mutex<BUS>, cs: CS, delay: D) -> Self {
+        Self { bus, cs, delay }
     }
 }
 
-impl<'a, BUS, CS> ErrorType for MutexDevice<'a, BUS, CS>
+impl<'a, BUS, CS> MutexDevice<'a, BUS, CS, super::NoDelay> {
+    /// Create a new MutexDevice without support for in-transaction delays.
+    ///
+    /// # Panics
+    ///
+    /// The returned device will panic if you try to execute a transaction
+    /// that contains one `Operation::Delay`.
```

Comment:
```suggestion
    /// that contains `Operation::DelayUs`.
```

Reply:
Just removing the "one" sounds a bit weird to me. Changed it to "that contains at least one Operation::DelayUs", hopefully that's fine?

Reply:
Saying "at least" sounds like a minimum _requirement_ to me, for example I'd be happy with "succeeds if it contains at least one Delay", but not "fails if it contains at least one Delay". I'd maybe say "that contains one or more Delay operations", but needing it to be plural makes it a bit awkward, so my first call would just be "that contains Delay".